include/deal.II/base/linear_index_iterator.h

```diff
@@ -262,8 +262,14 @@ class LinearIndexIterator
   /**
    * Inverse of operator==().
    */
-  bool
-  operator!=(const DerivedIterator &) const;
+  template <typename OtherIterator>
+  friend typename std::enable_if<
+    std::is_convertible<OtherIterator, DerivedIterator>::value,
+    bool>::type
+  operator!=(const LinearIndexIterator &left, const OtherIterator &right)
+  {
+    return left != right;
```

Comment:
I think this cannot compile in non C++20 mode because there is no rewrite overload for `operator!=`. Alaternatively, what about we just remove the `operator!=` altogether if `DEAL_II_HAVE_CXX20` is true?

Reply:
Yes, let's do that then.